Deployments fail unexpectedly with RDS identifier required error

Since the 29ht of April 2026 we are getting this error:
```
Error: 026-04-30T06:16:12.968] [ERROR] default - ╷
│ Error: Missing required argument
│ Error: Missing required argument
│ 
│   on main.tf.json line 1, in module.rds:
│    1: {"terraform":{},"module":{"rds":{"source":"terraform-aws-modules/rds/aws","version":"6.13.1"}}}
│ 
│ The argument "identifier" is required, but no definition was found.
```

It looks like the latest version of terraform has allowed some attributes to be specified in the module scope.

We are using the RDS module which `identifier` prop is required. But since cdktf is not maintained anymore the cli has not been updated since 10 months. So we get an error.

The solution is to fix the version of terraform that we use to the latest known working version: 1.14.9

--- a/.github/actions/deploy-infra/action.yml
+++ b/.github/actions/deploy-infra/action.yml
@@ -178,6 +178,8 @@ runs:
 
     - name: Setup Terraform
       uses: hashicorp/setup-terraform@v4
+      with:
+        terraform_version: '1.14.9'
 
     - name: Setup Node
       uses: actions/setup-node@v6

--- a/.github/actions/prepare-terraform/action.yml
+++ b/.github/actions/prepare-terraform/action.yml
@@ -16,6 +16,8 @@ runs:
   using: composite
   steps:
     - uses: hashicorp/setup-terraform@v4
+      with:
+        terraform_version: '1.14.9'
 
     - uses: actions/setup-node@v6
       with:

--- a/README.md
+++ b/README.md
@@ -19,7 +19,12 @@ Table of content:
 
 ### Locally
 
-- install [terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli)
+- install [terraform]
+  please install version 1.14.9 at most, later versions of terraform (starting at 1.15.0) are incompatible with cdktf which is not maintained anymore.
+  - Download version 1.14.9 from https://developer.hashicorp.com/terraform/install
+  - unzip locally
+  - move the terraform exec to /usr/bin/local (on macos) `sudo mv ~/Downloads/terraform_1.14.9_darwin_arm64/terraform /usr/local/bin/`
+  - ensure the version of terraform is at most 1.14.9 with `terraform --version`
 
 ### Infrastructure
 

